### PR TITLE
Add country codes to institutions/ endpoints

### PR DESCRIPTION
**Pull request by jasonjia93**
Country codes are now required for the /institutions/ API endpoints. This updates the client library to properly take a country_codes field

### CODE REVIEW

File: plaid/institutions.go [hunk 1]
```diff
@@ -122,25 +123,32 @@ type SearchInstitutionsResponse struct {
...
 func (c *Client) GetInstitutionByIDWithOptions(
 	id string,
+	countryCodes []string,
 	options GetInstitutionByIDOptions,
 ) (resp GetInstitutionByIDResponse, err error) {
 	if id == "" {
 		return resp, errors.New("/institutions/get_by_id - institution id must be specified")
 	}
 
+	if len(countryCodes) == 0 {
```
Comment (otherchen) [inline plaid/institutions.go:142]: Instead of doing one-off errors like this, I wonder if we should just rely on the API to return an API error saying the same thing so that we can keep all of our validation in one place. Wdyt?

File: plaid/institutions_test.go [hunk 2]
```diff
@@ -78,8 +82,8 @@ func TestGetInstitutionsByID(t *testing.T) {
 		t.Run(fmt.Sprintf("%#v", options), func(t *testing.T) {
 			// can't use the normal sandbox institution because it only returns the ItemLogins status.
 			institutionID := "ins_12"
-
-			instResp, err := testClient.GetInstitutionByIDWithOptions(institutionID, options)
+			countryCodes := []string{"US"}
+			instResp, err := testClient.GetInstitutionByIDWithOptions(institutionID, countryCodes, options)
 			assert.Nil(t, err)
```
Comment (otherchen) [inline plaid/institutions_test.go:87]: perhaps we can add a test to make sure the right error code is returned when we hit the api without an empty country codes array

**Review by otherchen — APPROVED** [review]
✔️

File: plaid/institutions.go (hunk 1 above)
Comment (jasonjia93) [reply plaid/institutions.go:142]: Yea I was kind of confused by this. I think it probably makes more sense to remove these and leave it up to the API to handle validation.
